backend/src/database/querys.rs

```diff
@@ -0,0 +1,173 @@
+use crate::{
+    database::dto::{AttestationResponse, Pagination},
+    error::AppError,
+};
+
+use sqlx::{postgres::PgQueryResult, Execute, FromRow, PgPool, Postgres, QueryBuilder};
+use uuid::Uuid;
+
+use super::dto::{AttestationRequest, Credential};
+
+pub async fn get_attestation_request_by_id(
+    attestation_request_id: &Uuid,
+    db_executor: &PgPool,
+) -> Result<AttestationResponse, AppError> {
+    sqlx::query_as!(
+        AttestationResponse,
+        "SELECT * FROM attestation_requests WHERE id = $1",
+        attestation_request_id
+    )
+    .fetch_one(db_executor)
+    .await
+    .map_err(AppError::from)
+}
+
+pub async fn get_attestations_count(db_executor: &PgPool) -> i64 {
+    sqlx::query_scalar!("SELECT COUNT (*) FROM attestation_requests")
+        .fetch_one(db_executor)
+        .await
+        .map_or(0, |count| count.unwrap())
+}
+
+pub async fn get_attestation_requests(
+    pagination: Pagination,
+    db_executor: &PgPool,
+) -> Result<Vec<AttestationResponse>, AppError> {
+    let query = build_pagination_query(pagination);
+    get_attestations(&query, db_executor)
+        .await
+        .map_err(AppError::from)
+}
+
+fn build_pagination_query(pagination: Pagination) -> String {
+    let mut query: QueryBuilder<Postgres> =
+        QueryBuilder::new("SELECT * FROM attestation_requests WHERE deleted_at IS NULL ");
+
+    if let Some(sorting) = pagination.sort {
+        query.push(format!("ORDER BY {} {} ", sorting[0], sorting[1]));
+    }
+
+    if let Some(offset) = pagination.offset {
+        query.push(format!("LIMIT {:?} OFFSET {:?}", offset[0], offset[1]));
+    }
+
+    query.build().sql().into()
+}
+
+async fn get_attestations(
+    query: &str,
+    db_executor: &PgPool,
+) -> Result<Vec<AttestationResponse>, sqlx::Error> {
+    let query_result = sqlx::query(query).fetch_all(db_executor).await;
```

Comment:
can't you just write `let attestation_rows: Vec<AttestationResponse> = sqlx::query(...).fetch_all(...).await?`

Reply:
`sqlx::query` returns a vector of PgRow. If I want to parse it directly into AttestationResponse I have tho use the macro call `query_as!` but this is not possible since I build the query dynamically and the macro has to know the query in compile time. I can not directly parse it into the struct (or at least I did not find a proper way to do that)